Phoenix rebranding

RhoSigma-QB64 opened the pull request:
**Note: This pull request draws the PR #129 by GeorgeMcGinn obsolete, as it practically overrides his changes, that PR should be closed first.**

-----
The changes in this PR will rebrand **QB64** to reflect we are developing the *Phoenix Edition*, a new fork of the software raised from the ashes of its predecessor. This new branding shall visibly distinguish us from other forks of this software out there.

To reach this goal, many of the occurrences of the term **QB64** are now called **QB64-PE** or fully written **QB64 Phoenix Edition** where applicable. Further file and folder names containing the term **qb64** are now renamed into **qb64pe**.

However, not everything was blindly renamed. Don't forget **QB64** is for itself also the name of the programming language, so I closely watched where the term is used in the context of the language and where it is used to denote our software product. I.e. only where the product is denoted, the term was renamed.

Also I've left untouched **any** internal uses of the **QB64** or **qb64** terms, which mainly means all kind of variable and constant names used in the software, such as C/C++ precompiler defines, make variables, and BASIC variables.

-----
To make reviewing the changes somewhat easier, I've splitted the whole process into several smaller (overseeable) steps. Matt, please take extra attention to the build script changes. Although I'm pretty sure I've caught everything important.

GeorgeMcGinn commented:
Actually, my changes in PR #129 isn’t in this pull request. My changes were two fold: 1) replace the icon for the desktop, and 2) Change the setup_lnx.sh so it does not overlay QB64 v2.0.2 or other versions people have on their Linux machines. The install replaces the .desktop file, replacing the program assigned to the QB64 icon to the PE edition. PE should be adding their own desktop file and corresponding icons.

RhoSigma-QB64 commented:
@GeorgeMcGinn this PR does replace #129.
- See commit labeled "Step 2" and compare it to your changes. The only difference is that I've also moved the icon image file from internal/source to ./source (internal/source is more or less a temporary snapshot of the last build) and changed that location in `setup_lnx.sh` accordingly.
- The requirements for windows are met in "Step 1" already, the icons here come either from `$EXEICON`, which is then deployed through a dynamically created resource file, or they come from the embedded images in libqb.cpp if no `$EXEICON` is used.

RhoSigma-QB64 commented:
@GeorgeMcGinn all checks on this PR have passed. If you wish, check out the LNX development build (Artifact) from here https://github.com/QB64-Phoenix-Edition/QB64pe/actions/runs/2851407195
